roles/gluster-secure-shared-storage/tasks/main.yml

```diff
@@ -0,0 +1,39 @@
+---
+# vim: set ts=2 sw=2 et :
+
+- name: Check for required variables
+  fail: msg="Variable {{ item }} must be defined to use this playbook"
+  when: item is undefined
+  with_items:
+  - cluster_master
+  - cluster_member_group
+
+- run_once: true
+  delegate_to: "{{ cluster_master }}"
+  block:
+  - name: Check if gluster_shared_storage is TLS enabled
+    shell: "gluster volume get gluster_shared_storage server.ssl |
+            grep server.ssl | awk '{ print $2}'"
+    changed_when: false
+    register: ss_server_tls_status
+
+  - name: Check if gluster_shared_storage clients are TLS enabled
+    shell: "gluster volume get gluster_shared_storage client.ssl |
+            grep client.ssl | awk '{ print $2}'"
+    changed_when: false
+    register: ss_client_tls_status
+
+  - name: Check if gluster_shared_storage is TLS allow restricted
+    shell: "gluster volume get gluster_shared_storage auth.ssl-allow |
+            grep auth.ssl-allow | awk '{ print $2}'"
+    changed_when: false
+    register: ss_tls_allow_status
+
+# False positive can occur here, when partial hosts are given, as the allow
+# list would not match, but the invoked play will do the right thing, by not
+# setting or changing any configuration.
+- name: Configure shared storage access restrictions
+  include_tasks: "restrict_shared_storage.yml"
+  when: (ss_server_tls_status.stdout != "on") or (ss_client_tls_status.stdout != "on") or
+        (ss_tls_allow_status.stdout != groups[hostvars[inventory_hostname].cluster_member_group]|
```

Comment:
Are there any issues around ordering of the host list? I don't know if groups[foo] returns a well defined order or not. I don't know that it would matter if we end up re-applying due to different sorting.
(Is this the false positive you mention?)

Reply:
Good catch!

Two things,
- Order is from the group, which is formed using the tag, so, will the list order be the same across runs? I guess depends on how the tags are processed and added to the list, and hence I would assume not.
- Running the play as the allow rules are different, because of the order, does no harm at present.

Ideally (if I were to write python, or maybe go ;) ), it would be easier to convert the output of ssl-allow to a list and do the comparison. I am not going to trust my Ansible-fu at this point to pull that off (unless it is needed of course).